crates/precompiles/src/tip20/mod.rs

```diff
@@ -641,6 +641,13 @@ impl TIP20Token {
         self.check_not_token_address(to)?;
         self.ensure_transfer_authorized(from, to)?;
 
+        // Only check access keys after Allegro Moderato hardfork
+        if self.storage.spec().is_allegro_moderato() {
+            // Check and update spending limits for access keys
+            let mut keychain = AccountKeychain::new();
+            keychain.authorize_transfer(from, self.address, amount)?;
+        }
```

Comment:
nit: we use this same snippet in a few places, it might make sense to encapsulate this into a function like `ensure_transfer_authorized()`.